### PR TITLE
walletconnect/Web3modal support

### PR DESCRIPTION
**Pull request by sirpy**
# Description
A wrapper around our web3provider that uses web3modal/wagmi/viem
use the exported useWeb3Modal to open the web3modal
it will wrap viem provider into an ethers web3provider and pass it to our web3provider.

This basically also means that you can use both usedapp and wagmi hooks.

### CODE REVIEW

File: lint-staged.config.js [hunk 1]
```diff
@@ -1,105 +1,3 @@
-const { ESLint } = require("eslint");
-const { opendirSync } = require("fs");
```
[the remark this answers is not in the document]
Comment (sirpy) [reply lint-staged.config.js:2]: i've added alternative more succint support eslint.
see master branch

[the remark this answers is not in the document]
Comment (L03TJ3) [reply lint-staged.config.js:2]: Ye @johnsmith-gooddollar can explain better as he wrote it

but what I am now missing to see...
it used to run on only changed files

doesn't it now not run on all files in mono?

[the remark this answers is not in the document]
Comment (sirpy) [reply lint-staged.config.js]: for all i know this is the kind of setup everyone uses.
from my testing it seems to run only on staged files.

**Review by L03TJ3 — APPROVED** [review]

**Review by L03TJ3 — CHANGES_REQUESTED** [review]
storybook is broken

**Review by L03TJ3 — APPROVED** [review]

**Pull request merged**